frameworks/hdfs/src/main/java/com/mesosphere/sdk/hdfs/scheduler/Main.java

```diff
@@ -128,4 +122,56 @@ private static String getHDFSUserAuthMappings(Map<String, String> env, String en
         String authMappings = new String(hdfsUserAuthMappingsBytes, "UTF-8");
         return authMappings;
     }
+
+    private static PodSpec getJournalPodSpec(ServiceSpec serviceSpec) {
+        // Journal nodes avoid themselves and Name nodes.
+        PlacementRule placementRule = new AndRule(
+                TaskTypeRule.avoid(JOURNAL_POD_TYPE), TaskTypeRule.avoid(NAME_POD_TYPE));
+
+        if (getPodSpec(serviceSpec, JOURNAL_POD_TYPE).getPlacementRule().isPresent()) {
```

Comment:
You could refactor the last 2/3 or each of these into a common function if you'd like.